internal/self_metrics/metrics.go

```diff
@@ -0,0 +1,167 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package self_metrics
+
+import (
+	"log"
+	"time"
+
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator"
+	timestamp "github.com/golang/protobuf/ptypes/timestamp"
+	metricpb "google.golang.org/genproto/googleapis/api/metric"
+	monitoredres "google.golang.org/genproto/googleapis/api/monitoredres"
+	monitoringpb "google.golang.org/genproto/googleapis/monitoring/v3"
+)
+
+type TimeSeries interface {
+	ToTimeSeries(instance_id, zone string) []*monitoringpb.TimeSeries
+}
+
+type Metric struct {
+	metric_type   string
```

Comment:
We use `camelCase` rather than `snake_case` for struct property and variable names usually.